packages/schematics/angular/environments/index.ts

```diff
@@ -72,15 +72,18 @@ function* generateConfigurationEnvironments(
   sourceRoot: string,
   projectName: string,
 ): Iterable<Rule> {
-  if (!buildTarget.builder.startsWith(AngularBuilder.Browser)) {
+  if (
+    buildTarget.builder !== AngularBuilder.Browser &&
+    buildTarget.builder !== AngularBuilder.Application
```

Comment:
Should probably put the `browser-esbuild` in here too so that we avoid false positives.